netstandard/tools/HandlePackageFileConflicts.cs

```diff
@@ -0,0 +1,323 @@
+                    return item2;
+                }
+            }
+
+            var fileVersion1 = GetFileVersion(sourcePath1);
+            var fileVersion2 = GetFileVersion(sourcePath2);
+
+            // if only one is missing version
+            if (fileVersion1 == null ^ fileVersion2 == null)
+            {
+                var nonVersion = fileVersion1 == null ? item1.ItemSpec : item2.ItemSpec;
+                Log.LogMessage($"{conflictMessage}. Could not determine a winner because {nonVersion} has no file version.");
+                return null;
+            }
+
+            if (fileVersion1 != fileVersion2)
+            {
+                if (fileVersion1 > fileVersion2)
+                {
+                    Log.LogMessage($"{conflictMessage}.  Choosing {item1.ItemSpec} because file version {fileVersion1} is greater than {fileVersion2}.");
+                    return item1;
+                }
+
+                if (fileVersion2 > fileVersion1)
+                {
+                    Log.LogMessage($"{conflictMessage}.  Choosing {item2.ItemSpec} because file version {fileVersion2} is greater than {fileVersion1}.");
+                    return item2;
+                }
+            }
+
+            var packageRank1 = GetPackageRank(item1);
+            var packageRank2 = GetPackageRank(item2);
+
+            if (packageRank1 < packageRank2)
+            {
+                Log.LogMessage($"{conflictMessage}.  Choosing {item1.ItemSpec} because package it comes from a package that is prefferred.");
+                return item1;
+            }
+
+            if (packageRank2 < packageRank1)
+            {
+                Log.LogMessage($"{conflictMessage}.  Choosing {item2.ItemSpec} because package it comes from a package that is prefferred.");
+                return item2;
+            }
+
+            Log.LogMessage($"{conflictMessage}.  Could not determine winner due to equal file and assembly versions.");
+            return null;
+        }
+
+        /// <summary>
+        /// Filters conflicts from original, maintaining order.
+        /// </summary>
+        /// <param name="original"></param>
+        /// <param name="conflicts"></param>
+        /// <returns></returns>
+        private ITaskItem[] RemoveConflicts(ITaskItem[] original, ICollection<ITaskItem> conflicts)
+        {
+            if (conflicts.Count == 0)
+            {
+                return original;
+            }
+
+            var result = new ITaskItem[original.Length - conflicts.Count];
+            int index = 0;
+
+            foreach(var originalItem in original)
+            {
+                if (!conflicts.Contains(originalItem))
+                {
+                    if (index >= result.Length)
+                    {
+                        throw new ArgumentException($"Items from {nameof(conflicts)} were missing from {nameof(original)}");
```

Comment:
this will not be caught, will dump stack to user. so should only happen if it indicates a bug in our code.